src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/caseupdate/services/CaseUpdaterTest.java

```diff
@@ -330,4 +117,60 @@ public void should_not_update_case_data_without_any_documents() {
         assertThat(exception).hasMessage("Missing scanned documents in exception record");
         verifyNoInteractions(addressExtractor);
     }
+
+    private void assertCaseData(SampleCase actualCaseData, SampleCase expectedCaseData) {
+        assertThat(actualCaseData)
+            .isEqualToComparingOnlyGivenFields(expectedCaseData,
+                "legacyId",
+                "firstName",
+                "lastName",
+                "dateOfBirth",
+                "bulkScanCaseReference"
+            );
+    }
+
+    private void assertScannedDocuments(
+        List<Item<ScannedDocument>> actualScannedDocuments,
+        List<Item<ScannedDocument>> caseScannedDocuments,
+        List<InputScannedDoc> exceptionRecordScannedDocuments
+    ) {
+        assertThat(actualScannedDocuments.get(0).value)
+            .isEqualToComparingOnlyGivenFields(caseScannedDocuments.get(0).value,
+                "type",
+                "scannedDate",
+                "deliveryDate",
+                "fileName",
+                "exceptionRecordReference",
+                "controlNumber"
+            );
+        assertThat(actualScannedDocuments.get(0).value.document)
+            .isEqualToComparingFieldByField(
+                caseScannedDocuments.get(0).value.document
+            );
+
+        assertExceptionRecordScannedDocument(
+            actualScannedDocuments.get(1).value,
+            exceptionRecordScannedDocuments.get(0)
+        );
+        assertExceptionRecordScannedDocument(
+            actualScannedDocuments.get(2).value,
+            exceptionRecordScannedDocuments.get(1)
+        );
```

Comment:
this definitely needs a comment?
Why are we comparing the second document from actual to first document in exception record and third one to second?

Reply:
Done